From 108c615f214efcd7999f916a63cb7da6108d9360 Mon Sep 17 00:00:00 2001
From: Fei Jiang <fei.jiang@intel.com>
Date: Fri, 29 Jun 2018 09:51:55 +0800
Subject: [PATCH 1574/2313] drm/i915/gvt: check pvmmio level with enable_pvmmio

Final pvmmio level is result of SOS's enable_pvmmio & UOS's enable_pvmmio.
Also need check detailed level when handling pvmmio mmio write.
SOS only patch.

Signed-off-by: Fei Jiang <fei.jiang@intel.com>
Reviewed-by: Min He <min.he@intel.com>
---
 drivers/gpu/drm/i915/gvt/handlers.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/handlers.c b/drivers/gpu/drm/i915/gvt/handlers.c
index 5c28e5c..7b54550 100644
--- a/drivers/gpu/drm/i915/gvt/handlers.c
+++ b/drivers/gpu/drm/i915/gvt/handlers.c
@@ -1312,7 +1312,7 @@ static int pvinfo_mmio_write(struct intel_vgpu *vgpu, unsigned int offset,
 				vgpu_vreg(vgpu, offset) = 0;
 				break;
 			}
-			vgpu_vreg(vgpu, offset) = !!data;
+			vgpu_vreg(vgpu, offset) = data & i915_modparams.enable_pvmmio;
 		} else {
 			vgpu_vreg(vgpu, offset) = 0;
 		}
@@ -1692,7 +1692,7 @@ static int elsp_mmio_write(struct intel_vgpu *vgpu, unsigned int offset,
 
 	execlist = &vgpu->execlist[ring_id];
 
-	if (VGPU_PVMMIO(vgpu)) {
+	if (VGPU_PVMMIO(vgpu) & PVMMIO_ELSP_SUBMIT) {
 		execlist->elsp_dwords.data[0] = elsp_data[0];
 		execlist->elsp_dwords.data[1] = elsp_data[1];
 		execlist->elsp_dwords.data[2] = elsp_data[2];
-- 
2.7.4

